### ISSUE TITLE
Navigation Block: prevent submenu items from merging

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#30164]
## Description

If you add multiple sub menu items and then hit backspace at the beginning of the second item, it merges with the first. 

## Step-by-step reproduction instructions

1. Add a navigation block
2. Start new. 
3. Add a Link Block.
4. Use the toolbar option to add a sub menu item under the link block you just created.
5. Do this once more.
6. Go to the very start of the second sub menu item you created and hit the backspace.
7. Find that the sub menu items merge together.

## Expected behaviour

I expect the items to stay separate. 

## Actual behaviour

They merged :) 

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/26996883/112238003-6d395400-8c09-11eb-8fbf-539d859ce24b.mov

## WordPress information
- WordPress version: 5.7
- Gutenberg version: 10.2.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1 Blocks

## Device information
- Device: Desktop
- Operating system: MacOS
- Browser: Chrome 88

**Comment by talldan** [comment]
Oh no, this one was implemented as a feature 😄 

Implemented in https://github.com/WordPress/gutenberg/issues/21743.

**Comment by talldan** [comment]
The button block does the same thing, and I think the two are very comparable and should work similarly.

I suggest we need design feedback on this, as it seems more like a general editor behavior that should be made consistent across blocks.

**Comment by annezazu** [comment]
Oh whoa interesting! I think if it were more reversible this would be less jarring :D

**Comment by talldan** [comment]
You can hit enter to split into two blocks again, but the original link url will be lost from the deletion of the block.

**Comment by annezazu** [comment]
Yeah :) To me, that feels less reversible. Plus once you do that there's no indication that a link is missing.

**Comment by jasmussen** [comment]
Speaking to the "submenu disappears" aspect (circling the cursor in the video), that should be fixed in #29869.

Edit: and if not by that, then by https://github.com/WordPress/gutenberg/pull/30169

**Comment by annezazu** [comment]
Looping back on old issues and returned to this one seeing the same behavior in 5.9 (just to get a current snapshot):

https://user-images.githubusercontent.com/26996883/152741063-8f7ce88e-1cf2-4bc9-ae6e-d37e390c3305.mov

I'd be curious to get more design thoughts on this as, to me, this is a pretty surprising thing to have happen. I can see this was originally implemented as a way to allow for an easier time deleting links. However, you can't easily undo this anymore (unless you hit undo) like you could before by separating the menu items back out as shown above. 

In my mind, nothing should happen upon taking this action.

**Comment by talldan** [comment]
>  However, you can't easily undo this anymore (unless you hit undo) like you could before by separating the menu items back out as shown above.

Yeah, that seems bad, seems like a regression of https://github.com/WordPress/gutenberg/issues/18208.

**Comment by annezazu** [comment]
Noting that this came up during the [FSE Outreach program's usability testing](https://make.wordpress.org/test/2022/06/20/fse-program-sign-up-for-usability-testing-by-june-24th/) with main menu items when deleting using 6.0 and Gutenberg 13.6:

https://user-images.githubusercontent.com/26996883/178381079-a486e616-56a4-42fa-ad7b-8a0a37c0eacd.mov

This felt like a broken experience to the participant.

**Comment by jordesign** [comment]
Noting that is still present when testing in WP6.3

**Comment by annezazu** [comment]
Still present in 6.7 RC1.

**Comment by getdave** [comment]
Confirmed. It would be good to get this one fixed. It will be related to Rich Text merging functionality.

**Comment by draganescu** [comment]
I would like to chime in to say I don't think this is a bug. It respects the flow within text blocks. Would it really be better if pressing backspace would have no effect?

**Comment by getdave** [comment]
I think the problem is if you have the block selected vs if you have a cursor within the label's richtext.

With the former backspace will delete the entire block.

With the later backspace will delete a character (unless it's empty in which case it merges with previous).

**Comment by jasmussen** [comment]
A good angle to think of it could be exactly what Dave described, deciding whether this is a writing flow feature where merging is desired, or whether navigation items should count as _non-text blocks_ like an image, where it gets deleted when pressing backspace. It seems the latter works well for Button blocks inside of the Buttons container, any reason navigation items can't work the same?

**Comment by scruffian** [comment]
I think this is the expected behaviour.
